lib/dock.rb

```diff
@@ -0,0 +1,35 @@
+class Dock
+  attr_reader :name, :max_rental_time, :current_rentals, :rental_history
+
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @current_rentals = []
+    @rental_history = []
+  end
+
+  def rent(boat, renter)
+    @current_rentals << {boat: boat, renter: renter}
+  end
+
+  def log_hour
+    @current_rentals.each do |rental|
+      boat = rental[:boat]
+      boat.add_hour if boat.hours_rented < @max_rental_time
+    end
+  end
+  
+  def return(boat)
```

Comment:
If you were storing your rentals as described above, you could do something like this:

```
rental = @current_rental[boat]
@rental_history << rental
@current_rentals.delete(boat)
```

This saves you from having to iterate through _all_ of the rentals each time a boat is returned.